src/pages/taskQueue/TaskQueueTable/index.tsx

```diff
@@ -0,0 +1,145 @@
+import { useMemo, useRef } from "react";
+import styled from "@emotion/styled";
+import Badge from "@leafygreen-ui/badge";
+import { LGColumnDef, useLeafyGreenTable } from "@leafygreen-ui/table";
+import { Body, Disclaimer } from "@leafygreen-ui/typography";
+import { useTaskQueueAnalytics } from "analytics";
+import { StyledRouterLink, WordBreak } from "components/styles";
+import { BaseTable } from "components/Table/BaseTable";
+import {
+  getVersionRoute,
+  getTaskRoute,
+  getUserPatchesRoute,
+} from "constants/routes";
+import { TaskQueueItem, TaskQueueItemType } from "gql/generated/types";
+import { formatZeroIndexForDisplay } from "utils/numbers";
+import { msToDuration } from "utils/string";
+
+interface TaskQueueTableProps {
+  taskQueue: Omit<TaskQueueItem, "revision">[];
+  loading: boolean;
+}
+
+const TaskQueueTable: React.FC<TaskQueueTableProps> = ({
```

Comment:
After deploying on beta, I see that the performance is noticeably slow when changing the distro compared to production. It's noticeable because the dropdown freezes in place and then there's a lag before the distro name gets updated. Is there any way we can improve it?

Reply:
I made some improvements and it should be better now.